mqtt/mqtt-edgehub/tests/edgehub_authorizer.rs

```diff
@@ -150,6 +162,9 @@ async fn auth_update_happy_case() {
         )
         .await;
 
+    // let authorizer update sink in...
+    time::delay_for(Duration::from_secs(1)).await;
```

Comment:
well, 1 sec is a lot of time to await... we are not under huge time pressure right now. can we do better than just waiting and hoping that update is processed?

Reply:
I'm not as sure with Rust, but our tests where we wait for something to happen in the dotnet side of things tends to lead to very difficult-to-debug tests that are flaky in pipelines

Reply:
about that... I'm open for concrete suggestions how to avoid delays. Atm, those tests are already flaky, b/c there is no way right now to reliably wait until the authorized is updated. At least adding delay only improves the current situation.

I know 1 sec is a lot (total 6 tests => 6 sec). But I've just cut ~15sec from each checkout task in the pipeline :smiley_cat:. I can give it another day or so to try to find a proper solution, but I'd like to push this PR with the policy engine tests out. @dmolokanov, @dylanbronson, what do you say?

Reply:
I think I might utilize `DummyAuthorizer` to send a signal when it has update. But I'd like to explore that in the next PR.

Reply:
Yep, sounds good. 
As an alternative we can add "callback" signal to `AuthorizerUpdate` and once update processed it sends signal to the receiver.

```rust
pub struct AuthorizerUpdate(Vec<IdentityUpdate>, oneshot::Sender<()>);

... 
fn update(&mut self, update: Box<dyn Any>) -> Result<(), Self::Error> {
        match update.downcast::<AuthorizerUpdate>() {
            Ok(update) => {
               // ... update authorizer here

              // signal that update has been processed
              update.1.send(());
            }
            Err(update) => {
                self.inner.update(update)?;
            }
        };

        Ok(())
    }
```